src/pages/storage/CustomVolumeSelectModal.tsx

```diff
@@ -0,0 +1,141 @@
+} from "context/loadCustomVolumes";
+import ScrollableTable from "components/ScrollableTable";
+
+interface Props {
+  project: string;
+  primaryVolume?: LxdStorageVolumeWithPool;
+  onFinish: (volume: LxdStorageVolumeWithPool) => void;
+  onCancel: () => void;
+  onCreate: () => void;
+}
+
+const CustomVolumeSelectModal: FC<Props> = ({
+  project,
+  primaryVolume,
+  onFinish,
+  onCancel,
+  onCreate,
+}) => {
+  const notify = useNotify();
+
+  const {
+    data: volumes = [],
+    error,
+    isLoading,
+  } = useQuery({
+    queryKey: [queryKeys.customVolumes],
+    queryFn: () => loadCustomVolumes(project),
+  });
+
+  if (error) {
+    notify.failure("Loading storage volumes failed", error);
+  }
+
+  const headers = [
+    { content: "Name" },
+    { content: "Pool" },
+    { content: "Content type" },
+    { content: "Used by" },
+    { "aria-label": "Actions", className: "actions" },
+  ];
+
+  const rows = volumes.map((volume) => {
+    return {
+      columns: [
+        {
+          content: (
+            <div className="u-truncate iso-name" title={volume.name}>
+              {volume.name}
+            </div>
+          ),
+          role: "cell",
+          "aria-label": "Name",
+        },
+        {
+          content: volume.pool,
+          role: "cell",
+          "aria-label": "Storage pool",
+        },
+        {
+          content: volume.content_type,
+          role: "cell",
+          "aria-label": "Uploaded at",
+        },
+        {
+          content: volume.used_by?.length,
+          role: "cell",
+          "aria-label": "Size",
+        },
+        {
+          content: (
+            <Button
+              onClick={() => onFinish(volume)}
+              dense
+              appearance={
+                primaryVolume?.name === volume.name &&
+                primaryVolume.type === volume.type &&
+                primaryVolume.pool == volume.pool
+                  ? "positive"
+                  : ""
+              }
+            >
+              Select
+            </Button>
+          ),
+          role: "cell",
+          "aria-label": "Actions",
+          className: "u-align--right",
+        },
+      ],
+    };
+  });
+
+  return (
+    <>
+      <div className="volume-table">
+        <ScrollableTable
+          dependencies={[volumes, rows, notify.notification]}
+          belowId="modal-footer"
+        >
+          <MainTable
+            headers={headers}
+            rows={rows}
+            paginate={30}
+            responsive
+            sortable
+            className="u-table-layout--auto"
+            emptyStateMsg={
+              isLoading ? (
+                <Loader text="Loading volumes..." />
+              ) : (
+                "No custom volumes found"
+              )
+            }
+          />
+        </ScrollableTable>
+      </div>
+      <footer className="p-modal__footer" id="modal-footer">
+        <Button className="u-no-margin--bottom" onClick={onCancel}>
```

Comment:
I think it makes sense here, in other cases the 2nd button is positive or negative, so a default button is fine in my opinion. Will take a note to discuss this with design in the future.

Reply:
This too can be a positive button in some cases. Actually, I just noticed that by default it gets the positive appearance at first load - namely, when we're still fetching the volumes. Is this expected and desired? Or should it have the default appearance by default and only take the positive one when we've fetched the list and there's no custom volume available?

Reply:
True, there is a little glitch here while loading. I think we shouldn't render the buttons while loading, will change it.

Reply:
Give it another default appearance before loading wouldn't solve it, as depending on the context, it can have both values after loading.

Reply:
Seeing it in action, though, makes me wonder if this is acceptable from a design perspective, because it's changing the modal's height - which is something that was mentioned many times before, that we should not do. Maybe visible but disabled buttons would be a better choice? Might be worth investigating this with @piperdeck

Reply:
Creation should be spaced now as expected.

Reply:
Not sure if we care, but it still looks bad at >1020px height  - this for example is with a ~1280px height:

![image](https://github.com/canonical/lxd-ui/assets/56583786/c937c8bc-2a90-48e3-aa7e-9d82dffeb90a)

Reply:
That's not what should be happening. I think we are getting closer. Now it should work there as well.

Reply:
Yes, I confirm that now it works well at any resolution. :+1: